### PR TITLE
[librdkafka] Add vcpkg-cmake-wrapper.cmake

### PR DESCRIPTION
**Pull request by PhoebeHui**
Related to https://github.com/microsoft/vcpkg/pull/8073, we don't need to set CPPKAFKA_RDKAFKA_STATIC_LIB in cppkafka.

@cbezault, could you help review?

### CODE REVIEW

File: ports/librdkafka/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        select_library_configurations(RdKafka)
+        
+        if("${_RdKafka_DEFS}" MATCHES "RdKafka_STATIC")
+            set(RdKafka_STATIC_LIBRARIES ${RdKafka_LIBRARIES})
+        else()
+            set(RdKafka_SHARED_LIBRARIES ${RdKafka_LIBRARIES})
+        endif()
+        
+        unset(_RdKafka_DEFS)
+        unset(_RdKafka_LINK_INTERFACE)
```
Comment (cbezault) [inline ports/librdkafka/vcpkg-cmake-wrapper.cmake]: You should clean up all variables that downstream consumers of the standard module wouldn't need. `TARGET_NAME` for example.

File: ports/librdkafka/vcpkg-cmake-wrapper.cmake [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        
+        list(APPEND RdKafka_LIBRARIES ${_RdKafka_LINK_INTERFACE})
+        list(APPEND RdKafka_LIBRARY ${_RdKafka_LINK_INTERFACE})
+        
+        select_library_configurations(RdKafka)
+        
+        if("${_RdKafka_DEFS}" MATCHES "RdKafka_STATIC")
+            set(RdKafka_STATIC_LIBRARIES ${RdKafka_LIBRARIES})
+        else()
+            set(RdKafka_SHARED_LIBRARIES ${RdKafka_LIBRARIES})
```
Comment (cbezault) [inline ports/librdkafka/vcpkg-cmake-wrapper.cmake]: Are these two variables needed by consumers? They're not listed as outputs of the standard module.

**Review by cbezault — CHANGES_REQUESTED** [review]
See #7331, it's very similar.

File: ports/librdkafka/vcpkg-cmake-wrapper.cmake [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        set(TARGETNAME RdKafka::rdkafka++)
+    endif()
+
+    if(TARGET ${TARGETNAME} AND NOT DEFINED RdKafka_INCLUDE_DIRS)
+        get_target_property(RdKafka_INCLUDE_DIRS ${TARGETNAME} INTERFACE_INCLUDE_DIRECTORIES)
+        set(RdKafka_INCLUDE_DIR ${RdKafka_INCLUDE_DIRS})
+        
+        get_target_property(_RdKafka_DEFS ${TARGETNAME} INTERFACE_COMPILE_DEFINITIONS)
+        
+        if("${_RdKafka_DEFS}" MATCHES "RdKafka_STATIC")
```
[the remark this answers is not in the document]
Comment (PhoebeHui) [reply ports/librdkafka/vcpkg-cmake-wrapper.cmake]: Updated, thanks for pointing out this!

File: ports/librdkafka/vcpkg-cmake-wrapper.cmake [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            get_target_property(RdKafka_LIBRARY_RELEASE ${TARGETNAME} IMPORTED_IMPLIB_RELEASE)       
+        else()
+            get_target_property(RdKafka_LIBRARY_DEBUG ${TARGETNAME} IMPORTED_LOCATION_DEBUG)
+            get_target_property(RdKafka_LIBRARY_RELEASE ${TARGETNAME} IMPORTED_LOCATION_RELEASE)
+        endif()
+        
+        get_target_property(_RdKafka_LINK_INTERFACE ${TARGETNAME} IMPORTED_LINK_INTERFACE_LIBRARIES_RELEASE) # same for debug and release
+        
+        list(APPEND RdKafka_LIBRARIES ${_RdKafka_LINK_INTERFACE})
+        list(APPEND RdKafka_LIBRARY ${_RdKafka_LINK_INTERFACE})
```
[the remark this answers is not in the document]
Comment (PhoebeHui) [reply ports/librdkafka/vcpkg-cmake-wrapper.cmake]: Updated, thanks!

File: ports/librdkafka/vcpkg-cmake-wrapper.cmake [hunk 5]
```diff
@@ -0,0 +1,42 @@
+include(SelectLibraryConfigurations)
+
+_find_package(RdKafka CONFIG)
```
[the remark this answers is not in the document]
Comment (PhoebeHui) [reply ports/librdkafka/vcpkg-cmake-wrapper.cmake]: Added, thanks!

File: ports/librdkafka/vcpkg-cmake-wrapper.cmake (hunk 1 above)
Comment (PhoebeHui) [reply ports/librdkafka/vcpkg-cmake-wrapper.cmake]: Updated, thanks for pointing out this!

File: ports/librdkafka/vcpkg-cmake-wrapper.cmake (hunk 2 above)
Comment (PhoebeHui) [reply ports/librdkafka/vcpkg-cmake-wrapper.cmake]: I removed them, thanks!

**Pull request merged**